extensions/interactions/GraphInput/GraphInput.js

```diff
@@ -694,12 +892,25 @@ oppia.factory('graphInputRulesService', [
   };
 
   return {
+    HasGraphProperty: function(answer, inputs) {
+      if (inputs.p == 'strongly_connected') {
+        return isStronglyConnected(answer);
+      } else if (inputs.p == 'weakly_connected') {
+        return isWeaklyConnected(answer);
+      } else if (inputs.p == 'acyclic') {
+        return isAcyclic(answer);
+      } else if (inputs.p == 'regular') {
+        return isRegular(answer);
+      } else {
+        return false;
```

Comment:
I'm not sure about that. It should never reach the `else` case, but even if we throw a frontend error, the player doesn't deal with that nicely (e.g. by bringing the user to an error page. It currently just appears to do nothing if it encounters an error.). None of the other frontend rules do input validation, either.